libflux/src/semantic/tests.rs

```diff
@@ -204,6 +204,636 @@ fn instantiation_1() {
     }
 }
 #[test]
+fn instantation_literals() {
+    test_infer! {
+        src: r#"
+            a = "Hello, World!"
+            b = 12
+            c = 18.5
+            d = 12h
+            e = 2019-10-31T00:00:00Z
+            f = /server[01]/
+        "#,
+        exp: &[
+            ("a", "forall [] string"),
+            ("b", "forall [] int"),
+            ("c", "forall [] float"),
+            ("d", "forall [] duration"),
+            ("e", "forall [] time"),
+            ("f", "forall [] regexp"),
+        ],
+    }
+}
+#[test]
+fn instantiate_string_expr() {
+    let src = r#"message = "Hello, ${name}!""#;
+    test_infer! {
+        env: &[
+            ("name", "forall [t0] t0"),
```

Comment:
We don't have types like this, at least not yet. Specifically we don't have value polymorphism, until we introduce polymorphic literals. So I would prefer we didn't include this specific test case as it's not really testing what we want to be testing.

I think what we want is something like this:
```Rust
test_infer! {
    env: &[
        ("name", "forall [] t0"),
    ],
    src: src,
    exp: &[
        ("name", "forall [] string"),
        ("message", "forall [] string"),
    ],
}
```
Note this test macro only prints out new additions to the environment. We'll need to update this macro to specify a "before" type environment, and an "after" type environment in order to support  this test case.

Reply:
Yes, this was something I was asking about. One thing that would be nice is if I could put anything in the expected array to see how it changed.

Reply:
As mentioned [here](https://github.com/influxdata/flux/pull/2102#issuecomment-551848574), after giving it some consideration, I think we should test cases like these indirectly via function expressions.